monai/utils/deprecated.py

```diff
@@ -10,19 +10,19 @@
 # limitations under the License.
 
 import inspect
-import re
 import warnings
 from functools import wraps
 from threading import Lock
 from types import FunctionType
 from typing import Optional
 
+from monai.utils.module import version_leq
+
 from .. import __version__
 
 __all__ = ["deprecated", "deprecated_arg", "DeprecatedError"]
 
 warned_set = set()
-warned_lock = Lock()
```

Comment:
looks like this variable was not in use? @ericspod

Reply:
No those were used by `warn_deprecated` to ensure warnings were issued only once.

Reply:
Hi @ericspod , maybe it's OK to always show the warning message when calling the API?
We can refer to other repos about handling this case.

Thanks.

Reply:
The issue is if you used a deprecated function in a loop or in a transform pipeline, it will get repeated a lot. We see this with warnings from Numpy or Scipy and it can be annoying.

Reply:
true, not sure if it could be addressed by filtering https://docs.python.org/3/library/warnings.html#the-warnings-filter

Reply:
It would require somehow reacting to when a warning comes up for the first time and filtering then, but you need some mechanism to detect that.